### PR TITLE
fix(linter): add v7 of typescript-eslint to peerDeps

### PR DESCRIPTION
**Pull request by batazor**
## Current Behavior

```
npm ERR! Could not resolve dependency:
npm ERR! peer @typescript-eslint/parser@"^6.13.2" from @nx/eslint-plugin@18.0.4
npm ERR! node_modules/@nx/eslint-plugin
npm ERR!   dev @nx/eslint-plugin@"*" from the root project
```

## Expected Behavior

success migration to storybook@v8 and no problem with upgrade other deps

## Related Issue(s)

Fixes #

### CODE REVIEW

**Comment by batazor** [comment]
@meeroslav can you review this MR?

File: packages/eslint-plugin/package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
   },
   "homepage": "https://nx.dev",
   "peerDependencies": {
-    "@typescript-eslint/parser": "^6.13.2",
+    "@typescript-eslint/parser": "^7.0.0",
```
Comment (seanparmelee) [inline packages/eslint-plugin/package.json:27]: this should probably allow both `^6.13.2` and `^7.0.0`

Comment (batazor) [reply packages/eslint-plugin/package.json]: Yeah, done

**Comment by seanparmelee** [comment]
@meeroslav @JamesHenry @leosvelperez any chance one of ya'll could take a look at this one? Thanks in advance!

**Review by meeroslav — APPROVED** [review]

**Comment by seanparmelee** [comment]
@batazor looks like the build is failing due to a commit message not following the convention: https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#commit-message-guidelines

**Pull request merged**